predicate/parser.go

```diff
@@ -91,9 +89,29 @@ func (p *parser) parseLogicalNode() (Node, error) {
 			if err != nil {
 				return *n, err
 			}
-			n.Children = append(n.Children, tr)
+			if n.Children[0] == nil {
+				n.Children[0] = tr
+			} else {
+				n.Children[1] = tr
+			}
 		case influxql.AND:
 			n.Operator = LogicalAnd
+			if n.Children[1] != nil {
```

Comment:
nit: could flip the comparison here and just continue, unindenting the main logic of this block

```go
if n.Children[1] == nil {
  continue
}
other stuffs...
```

Reply:
good idea